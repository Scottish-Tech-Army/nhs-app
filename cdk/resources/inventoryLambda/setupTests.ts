// eslint-disable-next-line no-global-assign
console = {
  ...console,
  log: jest.fn(),
  info: jest.fn(),
  debug: jest.fn(),
};

export const TEST_BOXES_TABLE_NAME = "dbtable-boxes";

process.env.BOXES_TABLE_NAME = TEST_BOXES_TABLE_NAME;

beforeEach(() => {
  jest.resetAllMocks();
});
