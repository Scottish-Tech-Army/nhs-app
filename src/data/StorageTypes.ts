export type ItemTemplate = {
  name: string;
  size?: string;
  description: string;
  location: string;
  photo: string;
  quantity?: number;
};

export type BoxTemplate = {
  boxTemplateId: string;
  name: string;
  items: ItemTemplate[];
  count: number;
};

export type StorageAreaTemplate = {
  storageAreaId: string;
  name: string;
  boxes: BoxTemplate[];
};

export type ItemContents = {
  name: string;
  size?: string;
  quantity: number;
};

export type BoxContents = {
  boxTemplateId: string;
  boxNumber: number;
  items: ItemContents[];
};

export type StorageAreaContents = {
  storageAreaId: string;
  boxes: BoxContents[];
};

export type ItemShoppingList = ItemContents;

export type BoxShoppingList = {
  boxTemplateId: string;
  boxNumber: number;
  name: string;
  items: ItemShoppingList[];
};
