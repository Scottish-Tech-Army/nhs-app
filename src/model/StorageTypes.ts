export type ItemTemplate = {
  name: string;
  size?: string;
  description: string;
  location: string;
  photo: string;
  quantity?: number;
};

export type ContainerTemplate = {
  containerTemplateId: string;
  name: string;
  containerType: string;
  items: ItemTemplate[];
  quantity?: number;
};

export type StorageAreaTemplate = {
  storageAreaId: string;
  name: string;
  containers: ContainerTemplate[];
  possibleContainerLocations?: string[];
};

export type StorageAreaGroupTemplate = {
  storageAreaGroupId: string;
  name: string;
  storageAreas: StorageAreaTemplate[];
};

export type MissingContainerItem = {
  name: string;
  size?: string;
  quantity: number;
  location?: string;
};

export type ContainerInputData = {
  checker: string;
  containerTemplateId: string;
  containerNumber?: number;
  storageAreaId: string;
  location?: string;
  name: string;
  missingItems: MissingContainerItem[];
  isFull: boolean;
};

export type ContainerData = ContainerInputData & {
  checkId: string;
  checkTime: string;
};
