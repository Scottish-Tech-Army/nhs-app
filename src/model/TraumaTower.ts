import {
  ItemContents,
  ItemTemplate,
  StorageAreaTemplate,
} from "./StorageTypes";

export const TRAUMA_TOWER_TEMPLATE: StorageAreaTemplate = {
  storageAreaId: "0",
  name: "Trauma Tower",
  boxes: [
    {
      name: "Trauma Chest Drain",
      boxTemplateId: "0",
      count: 6,
      items: [
        {
          name: "Blunt dissection chest drainage insertion pack",
          size: "28Fg",
          description:
            "Pack containing equipment for insertion of trauma chest drain",
          location: "Resus store XX",
          photo: "chestdrain.jpg",
        },
        {
          name: "Sterile gloves",
          size: "Small",
          description: "Small sterile gloves - 1 pair",
          location: "Resus store XX",
          photo: "smallsterilegloves.jpg",
        },
        {
          name: "Sterile gloves",
          size: "Medium",
          description: "Medium sterile gloves - 1 pair",
          location: "Resus store XX",
          photo: "mediumsterilegloves.jpg",
        },
        {
          name: "Sterile gloves",
          size: "Large",
          description: "Large sterile gloves - 1 pair",
          location: "Resus store XX",
          photo: "largesterilegloves.jpg",
        },
        {
          name: "Chest drain catheter",
          size: "28Fr",
          description: "Chest drain catheter - 28Fr",
          location: "Resus store XX",
          photo: "pleuralcatheter.jpg",
        },
        {
          name: "Chest drain catheter",
          size: "32Fr",
          description: "Chest drain catheter - 32Fr",
          location: "Resus store XX",
          photo: "pleuralcatheter.jpg",
        },
        {
          name: "Chest drain catheter",
          size: "36Fr",
          description: "Chest drain catheter - 36Fr",
          location: "Resus store XX",
          photo: "pleuralcatheter.jpg",
        },
        {
          name: "ChloraPrep applicator",
          description:
            "ChloraPrep applicator for cleaning skin for sterile procedure",
          location: "Resus store XX",
          photo: "chloraprep.jpg",
          quantity: 2,
        },
        {
          name: "Lidocaine 1%",
          size: "5ml / 50mg",
          description: "Local anaesthetic",
          location: "Resus store XX",
          photo: "lidocaine.jpg",
          quantity: 2,
        },
        {
          name: "Standard suture pack",
          size: "Standard",
          description: "Pack of equipment for suturing",
          location: "Resus store XX",
          photo: "standardsuture.jpg",
        },
        {
          name: "Mefix roll",
          size: "5cm x 10m",
          description: "Roll of mefix dressing retention tape",
          location: "Resus store XX",
          photo: "mefix.jpg",
        },
        {
          name: "Chest drain bottle",
          description: "Chest drain bottle for draining air and blood",
          location: "Resus store XX",
          photo: "chestdrainbottle.jpg",
        },
        {
          name: "Chest drain tubing",
          description:
            "Tubing for connecting chest drain catheter and chest drain bottle",
          location: "Resus store XX",
          photo: "chestdraintubing.jpg",
        },
        {
          name: "Sterile water (H20) bottle",
          size: "1000ml bottle",
          description: "Sterile water (H2O)",
          location: "Resus store XX",
          photo: "sterilewater.jpg",
        },
        {
          name: "Spencer wells forceps",
          size: "Straight 20cm",
          description:
            "Forceps for chest procedures including chest drain insertion, thoracostomies and thoracotomy",
          location: "Resus store XX",
          photo: "spencerwells.jpg",
        },
      ],
    },
  ],
};

export function getBoxTemplate(boxTemplateId: string) {
  return TRAUMA_TOWER_TEMPLATE.boxes.find(
    (boxTemplate) => boxTemplate.boxTemplateId === boxTemplateId
  );
}

export function getItemTemplate(
  boxTemplateId: string | undefined,
  itemNumberString: string | undefined
) {
  if (!boxTemplateId || !itemNumberString) {
    return undefined;
  }

  let itemNumber = -1;
  if (itemNumberString?.match(/^\d+$/)) {
    itemNumber = Number.parseInt(itemNumberString);
  }

  return getBoxTemplate(boxTemplateId)?.items[itemNumber];
}

export function getBoxName(boxTemplateName: string, boxNumber: number) {
  return `${boxTemplateName} - Box ${boxNumber}`;
}

export function getItemDisplayName(item: ItemTemplate | ItemContents) {
  let result = item.name;
  if (item.size) {
    result += ` (${item.size})`;
  }
  return result;
}
