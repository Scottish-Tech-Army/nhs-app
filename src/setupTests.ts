// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import localforage from "localforage";
import fetchMock from "jest-fetch-mock";

fetchMock.enableMocks();

jest.mock("localforage");

export const TEST_INVENTORY_API_ENDPOINT = "http://localhost:12345/api/";

process.env.REACT_APP_INVENTORY_API_ENDPOINT = TEST_INVENTORY_API_ENDPOINT;

// eslint-disable-next-line no-global-assign
console = {
  ...console,
  log: jest.fn(),
  info: jest.fn(),
  debug: jest.fn(),
  error: jest.fn(),
};

beforeEach(() => {
  jest.clearAllMocks();
  (localforage.getItem as jest.Mock).mockImplementation(() =>
    Promise.resolve(null)
  );
  (localforage.setItem as jest.Mock).mockImplementation(() =>
    Promise.resolve()
  );
  fetchMock.resetMocks();
});
