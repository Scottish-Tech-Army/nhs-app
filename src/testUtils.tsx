import React from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import type { AppStore, RootState } from "./data/store";
import boxContentsReducer, { initialState } from "./data/BoxContentsSlice";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProvider(
  component: React.ReactElement,
  {
    preloadedState = { boxContents: initialState },
    store = configureStore({
      reducer: { boxContents: boxContentsReducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {

  return {
    store,
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
}
