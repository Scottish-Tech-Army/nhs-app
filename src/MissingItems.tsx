import { Auth } from "@aws-amplify/auth";
import { format, parseISO } from "date-fns";
import React, { useEffect, useState } from "react";
import { ContainerData } from "./model/StorageTypes";
import {
  getContainerName,
  getContainerTemplate,
  getItemDisplayName,
  getStorageArea,
} from "./model/DataDefinitions";

import Navbar from "./Navbar";

export type FormValueType = {
  name: string;
  count: number;
};

const CONTAINERS_API_ENDPONT = `${process.env.REACT_APP_INVENTORY_API_ENDPOINT}containers`;

function MissingItems() {
  const [containers, setContainers] = useState<ContainerData[]>([]);
  const [loading, setLoading] = useState<boolean>(true);
  const partiallyFullContainers = containers.filter(
    (container) => !container.isFull
  );

  useEffect(() => {
    Auth.currentSession()
      .then((session) =>
        fetch(CONTAINERS_API_ENDPONT, {
          headers: {
            Authorization: `Bearer ${session.getIdToken().getJwtToken()}`,
          },
        })
      )
      .then((response) => response.json())
      .then((data) => {
        setContainers(data);
        setLoading(false);
      })
      .catch((error) => {
        console.error(error);
        setContainers([]);
      });
  }, []);

  function getDisplayTime(isoDateTime: string) {
    const timestamp = parseISO(isoDateTime);
    return format(timestamp, "EEE d/M/yyyy HH:mm");
  }

  function getContainerMissingItems(container: ContainerData, index: number) {
    const containerTemplate = getContainerTemplate(
      container.containerTemplateId
    );

    if (!containerTemplate) {
      console.error(
        "Container template not found: ",
        container.containerTemplateId
      );
      return null;
    }

    const storageArea = getStorageArea(container.storageAreaId);

    if (!storageArea) {
      console.error("Storage area not found: ", container.storageAreaId);
      return null;
    }

    return (
      <div key={index} className="container">
        <div>
          <h2>
            {storageArea.name}
            <br />
            {getContainerName(containerTemplate, container.containerNumber)}
          </h2>
          {!!container.location && (
            <div className="location">Location: {container.location}</div>
          )}
          <div className="checker">{`Checked: ${getDisplayTime(
            container.checkTime
          )} by ${container.checker}`}</div>
        </div>

        <div className="items">
          {container.missingItems.map((item, index) => (
            <div key={index} className="item">
              {`${item.quantity} x ${getItemDisplayName(item)}`}
            </div>
          ))}
        </div>
      </div>
    );
  }
  return (
    <div className="missing-items-list">
      <header>
        <h1>Missing Items</h1>
      </header>
      <main>
        {loading ? (
          <div>Fetching Items</div>
        ) : partiallyFullContainers.length ? (
          <div className="scroll">
            {partiallyFullContainers.map(getContainerMissingItems)}
          </div>
        ) : (
          <div className="nothing-to-replace">No Items</div>
        )}
      </main>
      <footer>
        <Navbar />
      </footer>
    </div>
  );
}

export default MissingItems;
