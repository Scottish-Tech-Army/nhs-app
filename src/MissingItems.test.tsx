/* eslint-disable testing-library/no-node-access */
import React from "react";

import { screen, waitFor } from "@testing-library/react";

import MissingItems from "./MissingItems";
import { renderWithProvider } from "./testUtils";
import { ContainerData } from "./model/StorageTypes";
import { TEST_INVENTORY_API_ENDPOINT } from "./setupTests";
import { Auth } from "@aws-amplify/auth";

jest.mock("@aws-amplify/auth");

const TIMESTAMP = "2022-06-23T09:18:06.324Z";
const DISPLAY_TIMESTAMP = "Thu 23/6/2022 10:18";

const UUID = "e5443b6c-4389-4119-a9c0-b7ad1f1eebc5";

const CONTAINERS: ContainerData[] = [
  {
    containerNumber: 2,
    containerTemplateId: "trauma-chest-drain",
    storageAreaId: "trauma-tower",
    checkId: UUID,
    checkTime: TIMESTAMP,
    checker: "Bob",
    location: "Resus 1b",
    isFull: false,
    missingItems: [
      { name: "Sterile gloves", quantity: 1, location: "Resus store XX", size: "Medium" },
      { name: "ChloraPrep applicator", quantity: 1, location: "Resus store XX" },
      { name: "Lidocaine 1%", quantity: 2, location: "Resus store XX", size: "5ml / 50mg" },
      { name: "Chest drain bottle", quantity: 1, location: "Resus store XX" },
    ],
    name: "Trauma Chest Drain",
  },
  {
    containerNumber: 4,
    containerTemplateId: "trauma-chest-drain",
    storageAreaId: "trauma-tower",
    checkId: UUID,
    checkTime: TIMESTAMP,
    checker: "Bob",
    isFull: true,
    missingItems: [],
    name: "Trauma Chest Drain",
  },
  {
    containerNumber: 1,
    containerTemplateId: "airway-trolley-2-drawer-c",
    storageAreaId: "airway-trolley-2",
    checkId: UUID,
    checkTime: TIMESTAMP,
    checker: "Joe",
    isFull: false,
    missingItems: [
      { name: "Guedel airway - Red", quantity: 1 },
      { name: "Nasopharyngeal Airway", quantity: 1, size: "7.0" },
    ],
    name: "Drawer C",
  },
];

describe("MissingItems", () => {
  beforeEach(() => {
    (Auth.currentSession as jest.Mock).mockImplementation(() => {
      return Promise.resolve({
        getIdToken: () => {
          return { getJwtToken: () => "test jwt token" };
        },
      });
    });
  });

  it("rendered a missing-items list page for filled store - no containers recorded", async () => {
    fetchMock.mockResponseOnce(JSON.stringify([]), { status: 200 });

    renderWithProvider(<MissingItems />, {
      initialRoutes: ["/missing-items"],
    });

    expect(
      screen.getByRole("heading", { name: "Missing Items" })
    ).toBeDefined();

    expect(await screen.findByText("No Items")).toBeDefined();

    await checkMissingItemsList([]);
  });

  it("rendered a missing-items list page - loading", async () => {
    fetchMock.mockResponseOnce("", { status: 500 });

    renderWithProvider(<MissingItems />, {
      initialRoutes: ["/missing-items"],
    });

    expect(
      screen.getByRole("heading", { name: "Missing Items" })
    ).toBeDefined();

    expect(screen.getByText("Fetching Items")).toBeDefined();

    await checkMissingItemsList([]);
  });

  it("rendered a missing-items list and all containers full", async () => {
    fetchMock.mockResponseOnce(
      JSON.stringify([
        {
          containerNumber: 4,
          containerTemplateId: "trauma-chest-drain",
          checkId: UUID,
          checkTime: TIMESTAMP,
          checker: "Bob",
          isFull: true,
          missingItems: [],
          name: "Trauma Chest Drain",
        },
        {
          containerNumber: 5,
          containerTemplateId: "trauma-chest-drain",
          checkId: UUID,
          checkTime: TIMESTAMP,
          checker: "Bob",
          isFull: true,
          missingItems: [],
          name: "Trauma Chest Drain",
        },
      ]),
      { status: 200 }
    );

    renderWithProvider(<MissingItems />, {
      initialRoutes: ["/missing-items"],
    });
    await waitFor(() => expect(fetchMock).toBeCalledTimes(1));
    expect(fetchMock).toBeCalledWith(
      TEST_INVENTORY_API_ENDPOINT + "containers",
      expect.objectContaining({
        headers: { Authorization: "Bearer test jwt token" },
      })
    );

    expect(
      screen.getByRole("heading", { name: "Missing Items" })
    ).toBeDefined();
    expect(await screen.findByText("No Items")).toBeDefined();

    await checkMissingItemsList([]);
  });

  it("rendered a missing-items list page for partially filled store - ie some items shown", async () => {
    fetchMock.mockResponseOnce(JSON.stringify(CONTAINERS), { status: 200 });

    const { container } = renderWithProvider(<MissingItems />, {
      initialRoutes: ["/missing-items"],
    });

    expect(
      screen.getByRole("heading", { name: "Missing Items" })
    ).toBeDefined();

    await checkMissingItemsList([
      {
        storageArea: "Trauma Tower",
        containerName: "Trauma Chest Drain - Box 2",
        location: "Resus 1b",
        checkNameAndDate: `Checked: ${DISPLAY_TIMESTAMP} by Bob`,
        items: [
          "1 x Sterile gloves (Medium) - Resus store XX",
          "1 x ChloraPrep applicator - Resus store XX",
          "2 x Lidocaine 1% (5ml / 50mg) - Resus store XX",
          "1 x Chest drain bottle - Resus store XX",
        ],
      },
      {
        storageArea: "Airway Trolley 2",
        containerName: "Drawer C",
        checkNameAndDate: `Checked: ${DISPLAY_TIMESTAMP} by Joe`,
        items: ["1 x Guedel airway - Red", "1 x Nasopharyngeal Airway (7.0)"],
      },
    ]);

    expect(container).toMatchSnapshot();
  });

  it("handle unknown containerTemplateId and storageAreaId", async () => {
    const unknownContainers: ContainerData[] = [
      {
        containerNumber: 2,
        containerTemplateId: "unknown",
        storageAreaId: "trauma-tower",
        checkId: UUID,
        checkTime: TIMESTAMP,
        checker: "Bob",
        location: "Resus 1b",
        isFull: false,
        missingItems: [{ name: "Sterile gloves", quantity: 1, size: "Medium" }],
        name: "Trauma Chest Drain",
      },
      {
        containerNumber: 1,
        containerTemplateId: "airway-trolley-2-drawer-c",
        storageAreaId: "unknown",
        checkId: UUID,
        checkTime: TIMESTAMP,
        checker: "Joe",
        isFull: false,
        missingItems: [{ name: "Guedel airway - Red", quantity: 1 }],
        name: "Drawer C",
      },
    ];

    fetchMock.mockResponseOnce(JSON.stringify(unknownContainers), { status: 200 });

    renderWithProvider(<MissingItems />, {
      initialRoutes: ["/missing-items"],
    });

    await waitFor(() => expect(fetchMock).toBeCalledTimes(1));

    expect(
      screen.getByRole("heading", { name: "Missing Items" })
    ).toBeDefined();

    await checkMissingItemsList([]);
  });

  it("can return to directory page", async () => {
    fetchMock.mockResponseOnce(JSON.stringify([]), { status: 200 });
    const { user, history } = renderWithProvider(<MissingItems />, {
      initialRoutes: ["/needed"],
    });

    expect(history.location.pathname).toEqual("/needed");

    await user.click(screen.getByRole("link", { name: "directory" }));

    expect(history.location.pathname).toEqual("/");
  });

  type ExpectedContainerContents = {
    storageArea: string;
    containerName: string;
    location?: string;
    checkNameAndDate: string;
    items: string[];
  };

  async function checkMissingItemsList(
    expectedContainers: ExpectedContainerContents[]
  ) {
    await waitFor(() => {
      const actualContainers = document.querySelectorAll("div.container")!;

      expect(actualContainers).toHaveLength(expectedContainers.length);
      actualContainers.forEach((actualContainer, index) => {
        const expectedContainer = expectedContainers[index];
        expect(actualContainer).toHaveTextContent(
          expectedContainer.storageArea
        );
        expect(actualContainer).toHaveTextContent(
          expectedContainer.containerName
        );
        if (expectedContainer.location) {
          expect(actualContainer).toHaveTextContent(
            expectedContainer.containerName
          );
        }
        expect(actualContainer).toHaveTextContent(
          expectedContainer.checkNameAndDate
        );

        const actualItems = actualContainer.querySelectorAll("div.item")!;
        expect(actualItems).toHaveLength(expectedContainer.items.length);
        actualItems.forEach((actualItem, index) => {
          const expectedItem = expectedContainer.items[index];
          expect(actualItem).toHaveTextContent(expectedItem);
        });
      });
    });
  }
});
