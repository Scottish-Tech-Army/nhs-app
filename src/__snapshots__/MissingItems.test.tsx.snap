// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissingItems rendered a missing-items list page for partially filled store - ie some items shown 1`] = `
<div>
  <div
    class="missing-items-list"
  >
    <header>
      <h1>
        Missing Items
      </h1>
    </header>
    <main>
      <div
        class="scroll"
      >
        <div
          class="container"
        >
          <div>
            <h2>
              Trauma Tower
              <br />
              Trauma Chest Drain - Box 2
            </h2>
            <div
              class="location"
            >
              Location: 
              Resus 1b
            </div>
            <div
              class="checker"
            >
              Checked: Thu 23/6/2022 10:18 by Bob
            </div>
          </div>
          <div
            class="items"
          >
            <div
              class="item"
            >
              1 x Sterile gloves (Medium) - Resus store XX
            </div>
            <div
              class="item"
            >
              1 x ChloraPrep applicator - Resus store XX
            </div>
            <div
              class="item"
            >
              2 x Lidocaine 1% (5ml / 50mg) - Resus store XX
            </div>
            <div
              class="item"
            >
              1 x Chest drain bottle - Resus store XX
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <div>
            <h2>
              Airway Trolley 2
              <br />
              Drawer C – Oxygenate by Any Means
            </h2>
            <div
              class="checker"
            >
              Checked: Thu 23/6/2022 10:18 by Joe
            </div>
          </div>
          <div
            class="items"
          >
            <div
              class="item"
            >
              1 x Guedel airway - Red
            </div>
            <div
              class="item"
            >
              1 x Nasopharyngeal Airway (7.0)
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer>
      <nav>
        <a
          aria-label="directory"
          class="navicon"
          href="/"
        >
          <svg>
            home.svg
          </svg>
          <div>
            Directory
          </div>
        </a>
        <a
          aria-current="page"
          aria-label="missing-items"
          class="navicon active"
          href="/missing-items"
        >
          <svg>
            checklist.svg
          </svg>
          <div>
            Missing Items
          </div>
        </a>
        <div
          aria-label="log out"
          class="navicon"
          id="auth-signout"
        >
          <svg>
            logout.svg
          </svg>
          <div>
            Log Out
          </div>
        </div>
      </nav>
    </footer>
  </div>
</div>
`;
